stakingModel/contracts/UBIScheme.sol

```diff
@@ -174,54 +151,58 @@ contract UBIScheme is AbstractUBI {
     }
 
     /* @dev Checks if the account has the privilege to claim. Only
-     * active accounts and new accounts that were been added the day
-     * before to the distribution list are eligible to claim.
+     * active accounts and accounts that were been added before to
+     * the pending list are eligible to claim. Adds inactive users
+     * to the pending list even if the user has not been fished.
      * @param account to update
      * @return A bool indicating if the account has the privilege to claim
      */
-    function _checkEntitlement(address account)
+    function _updateUserStatus(address account)
         private
         requireActive
-        onlyWhitelisted
         returns (bool)
     {
-        if(!checkActiveEntitlement(account)) { // the user is not active
-            if(checkNextDayEntitlement(account)) { // the user become to active
-                nextDayDistributionCount[currentDay] = nextDayDistributionCount[currentDay].sub(1);
-                activeUsersCount = activeUsersCount.add(1);
-                activeUsersAddresses[account] = true; // could not be fished
-                emit UserActivation(account, isNewUser(account)); // account and new or inactive
-            }
-            else { // the user has not last day entitlement
-                require(!nextDayDistributionAddresses[currentDay.add(1)][account].isEntitled, "has already added to the next day");
-                nextDayDistributionCount[currentDay.add(1)] = nextDayDistributionCount[currentDay.add(1)].add(1);
-                nextDayDistributionAddresses[currentDay.add(1)][account].isEntitled = true; // append the user to the next day distribution list
-                emit AddedToNextDay(account, isNewUser(account)); // account and new or inactive
+        if(isActiveUser(account)) {
+            require(!claimDay[currentDay].hasClaimed[account], "has already claimed");
+            return true;
+        }
+        if(pendingUsersAddresses[account].isPending) { // the user is in the pending list
```

Comment:
i dont think this is required here.
pending list i assume is the distribution list. the claim logic is as follows
1. if user isActive and lastClaimed<today then transfer his funds and update lastclaimed
2. if user is inActive then mark last claimed as today and add to pendinglist(ie distribution list)

when executing the distribution list claim, which simply calls the claim method in the name of the user
then condition 1 is now true, since distribution list is executed at the start of a new day. The user last claim is yesterday therefor the claim method will give him his funds.
if the user claims manually in the new day before distribution is called then his lastclaim will be updated and when called via distribution condition 1 will fail.

Reply:
updated followed by above
3a130df16ce6b41f40192c76921b58e63d2c5ccd